src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -865,6 +870,11 @@ await ReadCharAsync(cancellationToken).ConfigureAwait(false) != '.')
                     c = await ReadCharAsync(cancellationToken).ConfigureAwait(false);
                 }
 
+                if (response.Headers.ConnectionClose ?? false)
```

Comment:
Nope, I haven't measured perf of this.  

I should probably do something like the Transfer-Encoding thing you did.  My slight hesitation here is that at some point I'd like to revisit the parsing code generally (and the header management code as well) so I'd like to avoid workarounds that may go away in the future.

Let me think about it a bit... but I agree we should do something here before we merge.